circuit/src/gadgets/lt_eq_lookup.rs

```diff
@@ -0,0 +1,381 @@
+	circuit::{AssignedCell, Layouter},
+	plonk::{ConstraintSystem, Error, Selector, TableColumn},
+};
+
+/// Common config for the whole circuit
+#[derive(Clone, Debug)]
+pub struct MockCommonConfig {
+	/// Common config
+	pub common: CommonConfig,
+	/// Table column
+	pub table: TableColumn,
+}
+
+impl MockCommonConfig {
+	/// Create a new `MockCommonConfig` columns
+	pub fn new<F: FieldExt>(meta: &mut ConstraintSystem<F>) -> Self {
+		let common = CommonConfig::new(meta);
+		let table = meta.lookup_table_column();
+
+		Self { common, table }
+	}
+}
+
+/// Trait for an atomic chip implementation
+/// Each chip uses common config columns, but has its own selector
+pub trait MockChip<F: FieldExt> {
+	/// Output of the synthesis
+	type Output: Clone;
+	/// Gate configuration, using common config columns
+	fn configure(common: &MockCommonConfig, meta: &mut ConstraintSystem<F>) -> Selector;
+	/// Chip synthesis. This function can return an assigned cell to be used
+	/// elsewhere in the circuit
+	fn synthesize(
+		self, common: &MockCommonConfig, selector: &Selector, layouter: impl Layouter<F>,
+	) -> Result<Self::Output, Error>;
+}
+
+/// Chipset uses a collection of chips as primitives to build more abstract
+/// circuits
+pub trait MockChipset<F: FieldExt> {
+	/// Config used for synthesis
+	type Config: Clone;
+	/// Output of the synthesis
+	type Output: Clone;
+	/// Chipset synthesis. This function can have multiple smaller chips
+	/// synthesised inside. Also can returns an assigned cell.
+	fn synthesize(
+		self, common: &MockCommonConfig, config: &Self::Config, layouter: impl Layouter<F>,
+	) -> Result<Self::Output, Error>;
+}
+
+/// 1 << 252
+const N_SHIFTED: [u8; 32] = [
+	0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 16,
+];
+/// Numbers are limited to 252 to avoid overflow
+const NUM_BITS: usize = 252;
+/// Same number of bits as N_SHIFTED, since NUM + N_SHIFTED is the operation.
+const DIFF_BITS: usize = 253;
+
+/// Numbers range check uses 8-bit limb for lookup
+const K: usize = 8;
+/// Numbers range check uses 32(256 / 8) limbs for lookup
+const N: usize = 32;
+/// Numbers range check uses 4 bits, since it checks 252(256 - 4) bits long
+const S: usize = 4;
+
+#[derive(Clone, Debug)]
+/// Selectors for LessEqualChipset
+struct LessEqualConfig {
+	main: MainConfig,
+	lookup_range_check: RangeChipsetConfig,
+	bits_2_num_selector: Selector,
+	n_shifted_selector: Selector,
+}
+
+impl LessEqualConfig {
+	/// Constructs new config
+	fn new(
+		main: MainConfig, lookup_range_check: RangeChipsetConfig, bits_2_num_selector: Selector,
+		n_shifted_selector: Selector,
+	) -> Self {
+		Self { main, lookup_range_check, bits_2_num_selector, n_shifted_selector }
+	}
+}
+
+/// A chip for checking if number is in range
+struct LessEqualChipset<F: FieldExt> {
+	x: AssignedCell<F, F>,
+	y: AssignedCell<F, F>,
+	/// Bits of x and y and their difference
+	x_bits: [F; NUM_BITS],
+	y_bits: [F; NUM_BITS],
```

Comment:
If they are not used, they should be deleted

Reply:
Done https://github.com/eigen-trust/protocol/pull/169/commits/f5cb0f482c9f45b0327720226205386fe8a03613